enterprise_access/apps/content_assignments/api.py

```diff
@@ -452,23 +453,50 @@ def cancel_assignments(assignments: Iterable[LearnerContentAssignment]) -> dict:
     }
 
 
-def get_content_metadata_for_assignments(enterprise_catalog_uuid, assignments):
+def remind_assignments(assignments: Iterable[LearnerContentAssignment]) -> dict:
     """
-    Fetches (from cache or enterprise-catalog API call) content metadata
-    in bulk for the `content_keys` of the given assignments, provided
-    such metadata is related to the given `enterprise_catalog_uuid`.
+    Bulk remind assignments.
+
+    This is a no-op for assignments in the following states: [accepted, errored, canceled]. We only allow
+    assignments which are in the allocated state. Reminded and already-reminded assignments are bundled in
+    the response because this function is meant to be idempotent.
+
+
+    Args:
+        assignments (list(LearnerContentAssignment)): One or more assignments to remind.
 
     Returns:
-        A dict mapping every content key of the provided assignments
-        to a content metadata dictionary, or null if no such dictionary
-        could be found for a given key.
+        A dict representing reminded and non-remindable assignments:
+        {
+            'reminded': <list of 0 or more reminded or already-reminded assignments>,
+            'non-remindable': <list of 0 or more non-remindable assignments>,
+        }
     """
-    content_keys = sorted({assignment.content_key for assignment in assignments})
-    content_metadata_list = get_and_cache_catalog_content_metadata(enterprise_catalog_uuid, content_keys)
-    metadata_by_key = {
-        record['key']: record for record in content_metadata_list
-    }
+    from enterprise_access.apps.content_assignments.tasks import (  # pylint: disable=import-outside-toplevel
+        send_reminder_email_for_pending_assignment
+    )
+    remindable_assignments = set(
+        assignment for assignment in assignments
+        if assignment.state in LearnerContentAssignmentStateChoices.REMINDABLE_STATES
+    )
+    already_reminded_assignments = set(
+        assignment for assignment in assignments
+        if assignment.state == LearnerContentAssignmentStateChoices.REMINDED
```

Comment:
In a somewhat similar vein to Alex's comment below, I believe the assignment `state` should remain `allocated` even when there has been a reminder. Whether a reminder has been sent or not can be deduced from any associated successful reminder actions. The frontend "Assigned" table currently only intends to pull in assignment states of `allocated` or `errored` 😄 Additionally, the only column in the "Assigned" table that needs to know the latest reminder action is the "Recent action" column, which should already be set up to handle the display of the "Reminded: <last reminder date>" in that table cell.